Python/sort/monkey_sort.py

```diff
@@ -0,0 +1,42 @@
+"""
+Implementation of the MonkeySort (BogoSort) algorithm in Python.
+For implementing the this algorithm, we need to import a built-in module,
+named as 'random'.
+A demonstration on how the random module will be used:
+random_demo.py
+github.com/noviicee/Search_and_Sort/blob/main/Sorting%20Algos/Monkey%20Sort/random_demo.py
+The monkey sort works like what an actual monkey would do,
+if it is asked to sort a given set of cards.
+"""
+
+import random
+
+
+def is_sorted(data) -> bool:
+	"""
+	Takes in any data (here array), and returns a boolean expression:
+	True if the array is sorted
+	False if the array is not sorted
+	"""
+
+	return all(data[i] <= data[i + 1] for i in range(len(data) - 1))
+
+
+def monkey_sort(arr):
+	""" Takes in an array, returns the array in sorted form """
+
+	while not is_sorted(arr):
+		random.shuffle(arr)
+		print("Unordered lists by sorting the elements by shuffling them")
+		print(arr)
+	print("\nThe final sorted list:")
+	return(arr)
+
+if __name__ == "__main__" :
+    number_of_elements = int(input("Enter the number of elements : "))
+    print("Enter the elements:")
+    unsorted_array=list(map(int, input().split()))
+
+    # calling the monkey_sort function
+    sorted_array = monkey_sort(unsorted_array)
+    print(sorted_array)
```

Comment:
Add sample I/O and complexity of the Code

Reply:
@HarshCasper 
Providing a sample I/O for this code will result into a very long code because the program goes on printing random values of the list until it is sorted. Now it may get sorted in just 2 shuffles, or sometimes it may take many more shuffles to get sorted. 
Basically, the output cannot be guessed, since it contains randomly arranged elements.

Reply:
@HarshCasper 
I can add the complexity though, but it is never going to be constant. The average time complexity is O((N-1)* N!). The best-case occurs if the given array is already sorted, i.e., O(1) and the worst-case can lead to infinite time.

Reply:
> @HarshCasper
> Providing a sample I/O for this code will result into a very long code because the program goes on printing random values of the list until it is sorted. Now it may get sorted in just 2 shuffles, or sometimes it may take many more shuffles to get sorted.
> Basically, the output cannot be guessed, since it contains randomly arranged elements.

Yeah, I agree but still add a basic sample I/O where the array gets sorted after 3-4 shuffles. This is the basic template that we follow here to maintain code consistency.

Reply:
> @HarshCasper
> I can add the complexity though, but it is never going to be constant. The average time complexity is O((N-1)* N!). The best-case occurs if the given array is already sorted, i.e., O(1) and the worst-case can lead to infinite time.

Just mention the Worse time complexity along with the space complexity. Btw I guess the average time complexity should be O(N*N!)

Reply:
Okay doing that!
And yes the average time-complexity will be O(N*N!) XD